### PR TITLE
Speed up quoted printable encoding

### PR DESCRIPTION
**Pull request by bossek**
Encoding to quoted printable was quite slow in cases when calling
string:substr and subsequent lists:concat for inserting soft newline was
used too often. New implementation uses only fast head operations.

### CODE REVIEW

**Comment by bossek** [comment]
For comparing before/after performance I used following:

```
Mail = fun(Body) ->
 {<<"multipart">>, <<"alternative">>,
  [{<<"From">>, <<"test@example.org">>}, {<<"To">>, <<"test@example.org">>}, {<<"Subject">>, <<"test">>}, {<<"Mime-Version">>, <<"1.0">>}], #{},
  [{"text", "html", [{<<"Content-Type">>, <<"text/html; charset=\"utf-8\"">>}, {<<"Content-Transfer-Encoding">>, <<"quoted-printable">>}],
  #{content_type_params => [{<<"charset">>, <<"utf-8">>}], disposition => <<"inline">>}, Body}]}
end.

timer:tc(fun() -> mimemail:encode(Mail(unicode:characters_to_binary(string:copies(" ř", 100000))), []) end).
timer:tc(fun() -> mimemail:encode(Mail(binary:list_to_bin(string:copies("a a", 100000))), []) end).
```

Both calls takes seconds before improvement and hundreds/tens of milliseconds after.

File: src/mimemail.erl [hunk 1]
```diff
@@ -936,6 +932,11 @@ encode_quoted_printable(<<H, T/binary>>, Acc, L) ->
 	[A, B] = lists:flatten(io_lib:format("~2.16.0B", [H])),
 	encode_quoted_printable(T, [B, A, $= | Acc], L+3).
 
+insert_soft_newline([H | T], AfterPos) when AfterPos > 0 ->
+	[H | insert_soft_newline(T, AfterPos - 1)];
+insert_soft_newline(Str, 0) ->
+	[$\n, $\r, $= | Str].
```
Comment (seriyps) [inline src/mimemail.erl:938]: minor: I think you can do `"\n\r=" ++ Str` and compiler will be able to optimize it to what your code looks like now

**Review by seriyps — APPROVED** [review]
Nice, thanks!

**Comment by mworrell** [comment]
Good work, thank you!

**Comment by mworrell** [comment]
@seriyps Ready to be merged?

**Comment by seriyps** [comment]
@mworrell yes

File: src/mimemail.erl (hunk 1 above)
Comment (bossek) [reply src/mimemail.erl:938]: You are right! Should I change it before merge or keep it as it is?

Comment (seriyps) [reply src/mimemail.erl:938]: I'm fine as it is now.

Comment (bossek) [reply src/mimemail.erl:938]: OK, thanks!

**Comment by mworrell** [comment]
@bossek Merging now, thank you for your contribution!

**Comment by bossek** [comment]
@mworrell Glad to help :-)

**Pull request merged**